pkg/store/etcd.go

```diff
@@ -115,6 +110,30 @@ func (s *Etcd) Put(key string, value []byte) error {
 	return nil
 }
 
+// Create the entire path for a directory that does not exist
+func (s *Etcd) createDirectory(path string) error {
+	if _, err := s.client.CreateDir(normalize(path), 10); err != nil {
+		if etcdError, ok := err.(*etcd.EtcdError); ok {
+			if etcdError.ErrorCode != 105 { // Skip key already exists
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+// Periodically renew a key with an Ephemeral flag set
+func (s *Etcd) periodicRenew(key string, value []byte, ttl time.Duration) {
+	renew := time.NewTicker(40 * time.Second)
+	for _ = range renew.C {
+		if _, err := s.client.Set(key, string(value), uint64(ttl)); err != nil {
```

Comment:
Well cannot seem to avoid the CAS call here: there is no way with the client to only update the key's TTL without specifying a value, this is only do-able for a directory. So.. going back to the CAS solution..